xds/internal/test/xds_server_integration_test.go

```diff
@@ -58,94 +58,35 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
-func setupListenerResponse(respCh chan *fakeserver.Response, name string) {
-	respCh <- &fakeserver.Response{
-		Resp: &v2discoverypb.DiscoveryResponse{
-			Resources: []*anypb.Any{
-				{
-					TypeUrl: version.V2ListenerURL,
-					Value: func() []byte {
-						l := &v3listenerpb.Listener{
-							// This needs to match the name we are querying for.
-							Name: name,
-							ApiListener: &v3listenerpb.ApiListener{
-								ApiListener: &anypb.Any{
-									TypeUrl: version.V2HTTPConnManagerURL,
-									Value: func() []byte {
-										cm := &v3httppb.HttpConnectionManager{
-											RouteSpecifier: &v3httppb.HttpConnectionManager_Rds{
-												Rds: &v3httppb.Rds{
-													ConfigSource: &v3corepb.ConfigSource{
-														ConfigSourceSpecifier: &v3corepb.ConfigSource_Ads{Ads: &v3corepb.AggregatedConfigSource{}},
-													},
-													RouteConfigName: "route-config",
-												},
-											},
-										}
-										mcm, _ := proto.Marshal(cm)
-										return mcm
-									}(),
-								},
-							},
-						}
-						ml, _ := proto.Marshal(l)
-						return ml
-					}(),
-				},
-			},
-			TypeUrl: version.V2ListenerURL,
-		},
-	}
-}
-
-func setupBootstrapFile(t *testing.T, serverURI string) func() {
-	// Create a bootstrap file in a temporary directory.
-	tmpdir, err := ioutil.TempDir("", "xds-server-test*")
-	if err != nil {
-		t.Fatalf("failed to create tempdir: %v", err)
-	}
-	bootstrapContents := fmt.Sprintf(`
-		{
-			"node": {
-				"id": "ENVOY_NODE_ID",
-				"metadata": {
-				    "TRAFFICDIRECTOR_GRPC_HOSTNAME": "trafficdirector"
-			    }
-			},
-			"xds_servers" : [{
-				"server_uri": "%s",
-				"channel_creds": [
-					{ "type": "insecure" }
-				]
-			}]
-		}`, serverURI)
-	bootstrapFileName := path.Join(tmpdir, "bootstrap")
-	if err := ioutil.WriteFile(bootstrapFileName, []byte(bootstrapContents), os.ModePerm); err != nil {
-		t.Fatalf("failed to write bootstrap file: %v", err)
-	}
-
-	origBootstrapFileName := env.BootstrapFileName
-	env.BootstrapFileName = bootstrapFileName
-	t.Logf("Create bootstrap file at %s with contents\n%s", bootstrapFileName, bootstrapContents)
-	return func() { env.BootstrapFileName = origBootstrapFileName }
-}
-
 // TestServerSideXDS is an e2e tests for xDS use on the server. This does not
 // use any xDS features because we have not implemented any on the server side.
 func (s) TestServerSideXDS(t *testing.T) {
+	origV3Support := env.V3Support
```

Comment:
Would it make sense to merge this and `SetupBootstrapFile` into a `Setup()` function that takes the version as an input?

Reply:
I have changed the `SetupBootstrapFile` function to also accept a version, and based on that it generates appropriate bootstrap content.

But I think we should not put the logic of setting this environment variable into that because it should be possible to have a bootstrap file with `server_features = ["xds_v3"]`, but the environment variable is set to false, so the client ends up using v2. 

The server currently only exports the V3 API, but it should not be difficult to make it export the V2 API as well, and we can/should do that in the near future.

Reply:
SG